.ci/build.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# fail this script immediately if any command fails with a non-zero exit code
+set -e
+# fail on pipeline errors, e.g. when grepping
+set -o pipefail
+# fail on any unset environment variables
+set -u
+
+# based on https://gist.github.com/sj26/88e1c6584397bb7c13bd11108a579746?permalink_comment_id=4155247#gistcomment-4155247
+function retry {
```

Comment:
As a future enhancement I've been aiming to replace this helpers with native python code in a future iteration (at least in the repo where we first started using them). Maybe a TODO is worth to remind us about refactoring, or at least DRYing up the common bash functions somehow.

Reply:
Yeah, I agree that we should come up with a better, reusable approach in the future. Comment added.